lib/tasks/BackbeatTask.js

```diff
@@ -42,33 +35,40 @@ class BackbeatTask {
                 }
 
                 const now = Date.now();
-                if (now > (startTime + self.config.retryTimeoutS * 1000)) {
-                    log.error(
-                        `retried for too long to ${actionDesc}, giving up`,
+                const retriesMaxedOut = nbRetries >=
+                    this.retryParams.maxRetries;
+                const timeoutReached = now >=
+                    (startTime + this.retryParams.timeoutS * 1000);
+                // Give up if max retries reached or if timeout reached and
+                // the entry has been retried at least once
+                if (retriesMaxedOut || (timeoutReached && nbRetries > 0)) {
+                    log.error('giving up processing as retries are maxed out',
```

Comment:
Minor: could be useful to make the message conditional whether the timeout or number of retries was reached.

Reply:
I added that as log properties so that the message can be used for searching consistently and filtering the log properties `retriesMaxedOut` or `timeoutReached` would give the info where it timed out or had max retries reached.

Reply:
Good point re: searching logs.